Migrate "Shop parameters > Order settings > Statuses" page

Progi1984 opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Bye Bye Legacy & Hello Symfony
| Type?             | refacto
| Category?         | BO
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | Resolves #10619, resolves #10620, resolves #10621, resolves #21283
| How to test?      | This PR enables the Symfony pages for "Order States" pages<br><br>Nightly is : https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/5689492000

**:notebook: BC Breaks**
* Removed template `admin-dev/themes/default/template/controllers/statuses/helpers/form/form.tpl`
* Removed class `AdminStatusesController`


<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/27590)
<!-- Reviewable:end -->

Progi1984 committed:
Finish the migration of Order Statuses

matthieu-rolland reviewed (commented):
PR needs rebase, otherwise LGTM 👍

Progi1984 commented:
@NeOMakinG & @matthieu-rolland Rebased

jolelievre reviewed (changes requested):
Although I agree with the idea of this PR but it shouldn't be merged before it is fully finished You're saying in the PR's description that the modal will be added later, meaning the page is not fully migrated so it shouldn't be enabled yet

Either the modals are handled in a separate PR that needs to be merged first, or you can integrate them in this one it's ok as well But anyway the legacy page can't be removed and replaced until the new one is fully over

Progi1984 commented:
@jolelievre No soucis. I have already done a separate PR.

Blocked by #27606.

Hlavtox commented:
@Progi1984 Bro what is the status of this PR? There seems to be both modern and legacy code in the current develop?

Progi1984 commented:
@matthieu-rolland @jolelievre @PrestaShop/prestashop-maintainers : PR Rebased. The #27606 has been merged.

jolelievre reviewed (dismissed):
Thanks @Progi1984

marwachelly reviewed (commented):
Hello @Progi1984 ,

I tested the PR and I detected the following issues:

**The First issue:**

1. Go to shop parameters > statuses 
2. Check the box on the first line 
3. Bulk actions > delete selected
4. an exception is displayed

![image](https://user-images.githubusercontent.com/89604350/165480138-24158268-cbd6-4012-a9ff-b741d5e07a3c.png)

https://user-images.githubusercontent.com/89604350/165480197-65333585-1121-4d75-b92a-9372e13f629e.mp4


**The Second issue:**
I noticed that when entering an invalid characters in search by ID, the character is replaced by "." and the search is ignored which is not the case in other migrated page




https://user-images.githubusercontent.com/89604350/165483720-31ca1550-60a8-45d5-907e-c362482f5e99.mp4


**The Third issue**
Unecessary  space before fiedls in the listing page
![image](https://user-images.githubusercontent.com/89604350/165485639-351f3225-0977-4b06-b39a-3e29a52ff7f2.png)

**The Fourth issue:**

According to the wanted behavior from the  [issue](https://github.com/PrestaShop/PrestaShop/issues/15228)  The size of the label should be bigger which is not the case:
**_**with PR**_** 
![image](https://user-images.githubusercontent.com/89604350/165488483-9a585775-2388-4f05-b16c-eff877ddfe55.png)

_without PR_
![image](https://user-images.githubusercontent.com/89604350/165488801-158ba63f-addc-4fde-9185-5f7a6e924c25.png)

**The Fifth issue:**

A drop down when adding a new order status and clicking on the field status name


https://user-images.githubusercontent.com/89604350/165492786-9da84c17-8a8c-4889-af73-e53242a53086.mp4

**The sixth issue:**

Go to BO => Shop Parameters => Order Settings => Statuses page
Edit the Payment by check status
Check "Hide this status in all customer orders." option
Go to FO => order history of John Doe => exception

https://user-images.githubusercontent.com/89604350/165496428-6186fa9d-53a9-4e87-8a40-565178dc6221.mp4

**The seventh issue:**

1. Enable multistore and add a new shop
2. select the new shop
3. Go to BO => Shop Parameters => Order Settings => Statuses page
4. Add new order status or Edit an existing order status
5. check the box "Send an email to the customer when his/her order status has changed."
6. Preview Email template
7. see exception

https://user-images.githubusercontent.com/89604350/165508920-b2e8592e-f020-48e1-960c-113f38daf952.mp4


**The eighth issue**

1. Fresh install
2. Go to BO => Shop Parameters => Order Settings => Statuses page
3. Add new order status
4. Fill the name and color
5. Save **without** checking the box "Send an email to the customer when his/her order status has changed." then see the following issues

8.1- The column Email template from the status list is filled with "account" (It should be emty   (the case of the status delivered))
8.2- It is possible to change the status of the field "send email to the customer (mismatch with the [issue](https://github.com/PrestaShop/PrestaShop/issues/12105))

https://user-images.githubusercontent.com/89604350/165513727-4e148dcd-b2a9-4f5f-8db9-10454d6bb070.mp4


**The ninth issue:**

The search button is enabled when all search field are empty (It is not the case for other migrated pages)

https://user-images.githubusercontent.com/89604350/165518100-dbcab39e-c85e-445b-92fb-16788be64c68.mp4

**The tenth issue:**
When entering the status name some invalid characters in the help text are allowed:


https://user-images.githubusercontent.com/89604350/165523082-2320971f-db39-4beb-bcd7-84a785c3b60f.mp4

Could you please check?

Thanks,

Progi1984 commented:
@marwachelly 
* Issue 1 : Fixed
* Issue 2 : Fixed
* Issue 3 : It's not a bug. As decided with @MatShir and @jolelievre, we display only the checkbox when the row can be removed. 
* Issue 4 : Fixed
* Issue 5 : It comes from Chrome. Fixed
* Issue 6 : I don't think it comes from this PR. Have you check with develop branch ?
* Issue 7 : I didn't reproduce it.
* Issue 8.1 : Fixed
* Issue 8.2 : Fixed
* Issue 9 : Fixed
* Issue 10 : Fixed

jolelievre reviewed (dismissed):
Thanks @Progi1984

Progi1984 committed:
Status Grid : Disable/Enable Select All if needed & Change width of first column

marwachelly commented:
Hello,

Thanks @Progi1984  for your work. Many issues are fixed but the following problems stil exist :

**issue 1**
Unecessary space:
![image](https://user-images.githubusercontent.com/89604350/168564904-d6296165-fa96-4b06-9a40-ff826d5771c7.png)
Could you please compare with other migrated page (for exemple page order)
![image](https://user-images.githubusercontent.com/89604350/168564763-26c5fd26-bd2a-4f4c-8356-47dc7c1d3677.png)

**issue 2**
Exception when trying to add a Return statuses with a many characters
https://user-images.githubusercontent.com/89604350/168568063-27119000-6114-42fe-a185-c9e9601939cf.mp4

 when trying to add a new order statuses with a more then 64 chars I faced the following issue:
 

1.  an exception when trying to save
2.  An invalid statues is added
3.  an exception when trying to delete the invalid order staus added 

 
 https://user-images.githubusercontent.com/89604350/168572473-987cce7a-9afd-4ff2-ab5d-3e0fe8e8bb52.mp4
 
**issue 3**
 drop down when clicking to add new order status
 
 **issue 4**
 
1. Fresh install
2. Go to BO => Shop Parameters => Order Settings => Statuses page
3. Add new order status
4. Fill the name and color
5. Save **without** checking the box "Send an email to the customer when his/her order status has changed." then see the following issues
6. Edit the status

8.1- The column Email template from the status list is filled with "account" (It should be emty   (the case of the status delivered))
8.2- It is possible to change the status of the field "send email to the customer (mismatch with the [issue](https://github.com/PrestaShop/PrestaShop/issues/12105))

https://user-images.githubusercontent.com/89604350/168578083-80bcaa02-a59f-4828-ba0d-949ca77735ab.mp4

**issue 5**
1. Enable multistore and add a new shop
2. select the new shop
3. Go to BO => Shop Parameters => Order Settings => Statuses page
4. Add new order status or Edit an existing order status
5. check the box "Send an email to the customer when his/her order status has changed."
6. Preview Email template
7. see exception

https://user-images.githubusercontent.com/89604350/168605545-eb1e317f-0df9-4677-bd48-e681db09390b.mp4

Could you please check? 

Thanks,

marwachelly reviewed (commented):
Hello,

Thanks @Progi1984  for your work. Many issues are fixed but the following problems stil exist :
**issue 1**
Unecessary space:
![image](https://user-images.githubusercontent.com/89604350/168564904-d6296165-fa96-4b06-9a40-ff826d5771c7.png)
Could you please compare with other migrated page (for exemple page order)
![image](https://user-images.githubusercontent.com/89604350/168564763-26c5fd26-bd2a-4f4c-8356-47dc7c1d3677.png)

**issue 2**
Exception when trying to add a Return statuses with a many characters
https://user-images.githubusercontent.com/89604350/168568063-27119000-6114-42fe-a185-c9e9601939cf.mp4

 when trying to add a new order statuses with a more then 64 chars I faced the following issue:
 
 an exception when trying to save
 An invalid statues is added
 an exception when trying to delete the invalid order staus added 
 
 https://user-images.githubusercontent.com/89604350/168572473-987cce7a-9afd-4ff2-ab5d-3e0fe8e8bb52.mp4
 
 
**issue 3**
 
1. Fresh install
2. Go to BO => Shop Parameters => Order Settings => Statuses page
3. Add new order status
4. Fill the name and color
5. Save **without** checking the box "Send an email to the customer when his/her order status has changed." then see the following issues
6. Edit the status:

- The column Email template from the status list is filled with "account" (It should be emty   (the case of the status delivered))
- It is possible to change the status of the field "send email to the customer (mismatch with the [issue](https://github.com/PrestaShop/PrestaShop/issues/12105))

https://user-images.githubusercontent.com/89604350/168578083-80bcaa02-a59f-4828-ba0d-949ca77735ab.mp4

**issue 4**
1. Enable multistore and add a new shop
2. select the new shop
3. Go to BO => Shop Parameters => Order Settings => Statuses page
4. Add new order status or Edit an existing order status
5. check the box "Send an email to the customer when his/her order status has changed."
6. Preview Email template
7. see exception

https://user-images.githubusercontent.com/89604350/168605545-eb1e317f-0df9-4677-bd48-e681db09390b.mp4

Could you please check? 

Thanks,

Progi1984 committed:
Fixes the constraint of length for a new Order Return Status

Progi1984 committed:
Fixes the size of the first column in Grid

Progi1984 committed:
Fixes edit of Order Status with template & hasEmail options

Progi1984 committed:
Fixed preview email in MultiShop

Hlavtox commented:
@marwachelly Issue 1 is not related to this PR and not really an issue. 

There is not as much data as in other tables, that's why there is space and the browser sizes the columns somehow evenly. If you limit the width of columns, you are asking for trouble and you can never know what they will contain. :-)

Progi1984 commented:
Ping @jolelievre & @NeOMakinG for review

NeOMakinG reviewed (dismissed):
front lgtm

marwachelly reviewed (commented):
Hello @Progi1984 
**issue 1**
Still not fixed especially for return status tab
Unecessary space:
![image](https://user-images.githubusercontent.com/89604350/168564904-d6296165-fa96-4b06-9a40-ff826d5771c7.png)
Could you please compare with other migrated page (for exemple page order)
![image](https://user-images.githubusercontent.com/89604350/168564763-26c5fd26-bd2a-4f4c-8356-47dc7c1d3677.png)



**issue 2**
Exception when trying to add a Return statuses with a many characters => Fixed 


 But when trying to add a new order statuses with a more then 64 chars I faced the following issue:
 

-  an exception when trying to save
-  An invalid statues is added
-  an exception when trying to delete the invalid order staus added 

 
 https://user-images.githubusercontent.com/89604350/168572473-987cce7a-9afd-4ff2-ab5d-3e0fe8e8bb52.mp4
 
 **issue 3** 
Issue is well fixed

https://user-images.githubusercontent.com/89604350/179793298-eab21d7a-486e-4d9f-aad6-ddfb37137b57.mp4

**issue 4** 
Issue is well fixed


https://user-images.githubusercontent.com/89604350/179795497-acf3103c-acfe-4de8-bcdf-1307a1f4fdfa.mp4


Thank you for your work. Issue 3 and issue 4 are well fixed. Could you please check the issue 1 and issue 2?

Progi1984 committed:
Fixes max length of the OrderState in Form

Progi1984 commented:
@marwachelly Thanks for the QA.

The issue 2 has been fixed.

For the issue 1, I would like the advice from @PrestaShop/product-team : The grid take the 100% width, cell auto adjusts on the content of data. These grid are less filled than the order grid so the width of cells are taller than those of order grid.

zuk3975 commented:
> For the issue 1, I would like the advice from https://github.com/orgs/PrestaShop/teams/product-team : The grid take the 100% width, cell auto adjusts on the content of data. These grid are less filled than the order grid so the width of cells are taller than those of order grid.

I agree with you and I think it shouldn't be in the scope of this PR (but rather a dedicated PR for global grids styling, after it is actually accepted as an issue by UX team), because if you look at other grids it acts the same (lets take for example brands - screenshot bellow)
![Screenshot from 2022-07-20 15-11-57](https://user-images.githubusercontent.com/31609858/179978958-ce259fd9-21dc-4b37-8c6f-b5877d1a0476.png)

zuk3975 commented on src/Core/Domain/OrderReturnState/OrderReturnStateSettings.php:
```diff
@@ -24,12 +24,14 @@
 
-header('Expires: Mon, 26 Jul 1997 05:00:00 GMT');
-header('Last-Modified: '.gmdate('D, d M Y H:i:s').' GMT');
+declare(strict_types=1);
```
I don't understand why git shows these files changes as if it would be the same file, instead of showing that index.php was removed separately :thinking:

zuk3975 commented on admin-dev/themes/new-theme/scss/components/_grid.scss:
```diff
@@ -309,10 +309,6 @@ table {
       td[data-column-id^="id_"] {
         max-width: 8rem;
-
-        input {
```
Are you sure about this change?
As I see the id column size was made smaller intentionally by @NeOMakinG in https://github.com/PrestaShop/PrestaShop/pull/24952

Progi1984 replied:
Yes. The size of columns will be worked on a new PR.

Hlavtox commented:
@marwachelly Regarding issue 1 - I think you could accept my word as a frontend maintainer and stop blocking this PR for it, as specified here: https://github.com/PrestaShop/PrestaShop/pull/27590#issuecomment-1128625874

There is no problem in @Progi1984's work.

Table stretching to maximum width is a normal behavior. If you limit one column, other one will stretch, unless you limit the width of the whole table.

MatShir commented:
Thanks @marwachelly highlighting the issue 1, the table view is not exactly the same before the migration. But it's not an issue that is blocking the PR. But from an esthetic point of view, we could define the max width. It can be done in another issue/PR.

Progi1984 commented:
@MatShir So no need to "Waiting for PM" label. Isn't it ?

marwachelly reviewed (dismissed):
Hello @Progi1984,

Thank you for your work! The issue is well fixed. The manual tests are ok. Just some [automatic tests](https://github.com/marwachelly/testing_pr/actions/runs/2782585481) are failed and we need a feedback from our @PrestaShop/qa-automation 



https://user-images.githubusercontent.com/89604350/182586959-69113f2c-77c8-4b56-a390-0c1598b60d4b.mp4


@PrestaShop/qa-automation  Could you please check? 

Thanks,

zuk3975 commented:
> Hello @Progi1984,
> 
> Thank you for your work! The issue is well fixed. The manual tests are ok. Just some [automatic tests](https://github.com/marwachelly/testing_pr/actions/runs/2782585481) are failed and we need a feedback from our @PrestaShop/qa-automation
> 
>  Untitled_.Aug.3.2022.11_02.AM.mp4 
> @PrestaShop/qa-automation Could you please check?
> 
> Thanks,

@marwachelly in such cases you shouldn't probably approve the PR (idk for real, but it seems so for me)? As there is only one QA check required for PR to be merged, so even though your comment states that something fails, this PR could have been merged already with those failures :thinking: (or else there should be yet another TAG/Approval introduced for automation team if its somekind of separate thing).. Just concerned though, no bad feelings :v: :smile:

boubkerbribri commented:
Hello @zuk3975 
It's a special case for migration PR. It's validated by QA, but needs some updates on UI tests.
So to do it on the same pull request, we follow this process (not documented yet) :
- Validate manually the PR.
-  if UI tests fails, we ping qa-automation team to update tests.
-  when it's done, we push on the same PR.
- UI tests only will be reviewed.
- when it approuved, we add the QA ✔️ to it will be merged (without QA testing again)

In some cases, UI tests are updated/fixed by the author, then passing it to qa-automation team will not be necessary.

Why this process? Because it's hard to follow all PR like this after merging, and it's a regression if a PR is breaking tests.

hibatallahAouadni reviewed (dismissed):
Hello @Progi1984 

As dsicussed on Slack with @Robin-Fischer-PS and @boubkerbribri, I encounter 2 weird behavior in `Add new order return status` and `Add new order status` forms:

1. the color field doesn't have any outline and looks like there's no field
2. it's impossible to type the color name :disappointed: I can only paste the color name

https://user-images.githubusercontent.com/23308427/182879413-418276d9-1236-4daf-8bb0-bbae69ee8fa0.mp4

Question: why when we change the color value, the field's background doesn't change along?
Waiting for your feedback.

Thanks!

Progi1984 committed:
Colorpicker : Fixed the color of input

Progi1984 commented:
@hibatallahAouadni (and @NeOMakinG) I fixed the problem : the color change with the color picker click, the color identifier is not displayed (like blockreassurance).

@PrestaShop/prestashop-maintainers & @PrestaShop/committers Need a (maybe last) review when you will have some free time : only change since the last time is the commit d1b54143585721f895ee0b258cb0cfce3b93e084. Could you remove the label "Waiting for author" ?

NeOMakinG reviewed (dismissed):
🍬 

I hope you checked how it reacts in case you select the white or black color, is the input text human readable?

Hlavtox commented:
`$font_color = Tools::getBrightness($hex) > 128) ? '#000' : '#fff'`

Progi1984 commented:
@NeOMakinG I reproduce the behavior in blockreassurance, like @hibatallahAouadni said me. So the text of the input is hidden.

Progi1984 commented:
Someone can put the label "Waiting for QA" ? There is one approval of a maintainer, and one approval of a committer.

boubkerbribri commented:
ping @nesrineabdmouleh , if QA :heavy_check_mark: we should fix UI tests for the migrated page.

Progi1984 commented:
@boubkerbribri May be I could work on it next week :wink:

hibatallahAouadni commented:
Hello @Progi1984 

As disccussed with @NeOMakinG and as he mentioned above:
> 🍬 
>
> I hope you checked how it reacts in case you select the white or black color, is the input text human **readable**?

That's not normal neither for this page nor to the module, this behavior is totally wrong! As you can see in the attached screen record below:
I'm really sorry to ask to redo it again :disappointed: :pray: 

https://user-images.githubusercontent.com/23308427/188645596-3affd32d-4440-4943-bd49-09b4f566fb53.mp4

Ping @PrestaShop/ux-team & @PrestaShop/product-team what kind of input we should use for this color picker :pray: 
Should we add just some border and padding, see the attached screenshot below:
```
border: 1px solid rgb(187, 205, 210);
padding: 0px 5px;
```

![image](https://user-images.githubusercontent.com/23308427/188655681-16362276-002f-4c31-bdb6-0d2ec3f21bc2.png)

![image](https://user-images.githubusercontent.com/23308427/188655702-695238b6-85a5-4109-9713-0d0e4c610e78.png)

Waiting for your feedback.

Thanks

Progi1984 commented:
Blocked by https://github.com/PrestaShop/prestashop-ui-kit/issues/196

kpodemski commented:
@PrestaShop/product-team can't we just use Symfony's color picker? 🤔

Hlavtox commented:
Yes, thats what I also thought.

MatShir commented:
I think we can use it, until design comes back. It is sad to see this PR blocked for this.
Since it is for the v9, the page might need to be behind a feature flag. I'm wondering if keep the pages hidden is enough ? 🤔 
I need tech maintainer for this decision @eternoendless @kpodemski

Hlavtox commented:
We should just use normal html color input that symfony provides us. https://symfony.com/doc/4.4/reference/forms/types/color.html

It's supported in all modern browsers - https://caniuse.com/input-color

eternoendless commented:
> Since it is for the v9, the page might need to be behind a feature flag. I'm wondering if keep the pages hidden is enough ?

It looks like this PR deletes the old page. It should keep it and allow switching to it using feature flags.

Hlavtox commented:
@PrestaShop/committers Guys what about his PR, it's blocked because of a request to add a color picker )https://github.com/PrestaShop/prestashop-ui-kit/issues/196), where the UX responsible person is delivering a design since September.


@leemyongpakvn Used standard Symfony color picker in https://github.com/PrestaShop/example-modules/pull/106 and it works just fine.

Progi1984 commented:
We wait some news about @ElodieOS .

MatShir commented:
We can use this design as it is now ! But it has to be hidden behind a feature flag :)

ElodieOS commented:
Hello @Progi1984 ! 
The color picker is ready but I have to wait for it to be reviewed. 

It's better to use another design for now, as @MatShir said, because I don't know when it will be reviewed (although I think it's almost ready). 

Sorry for the wait 😕.

zuk3975 reviewed (dismissed):
so this PR still needs a rebase and this 
> It looks like this PR deletes the old page. It should keep it and allow switching to it using feature flags.

eternoendless commented:
If the page is merged for v9 then I guess we can still remove the old page 🤷‍♂️ 
As long as there are no functional regressions.

Progi1984 committed:
Functional Tests : Fixed UI Tests

Progi1984 commented:
@hibatallahAouadni No need to Feature Flag, it's for V9.

> We can use this design as it is now !
The colorpicker will be upgraded on all the project in the future. (I will check it).

Progi1984 commented:
@PrestaShop/committers The PR is ready for review. Nightly is in :red_circle: because ps_facetedsearch is not updated on develop.  Please review when you have some availability.

@PrestaShop/qa-automation : UI Tests are fixed so when the PR will be merged, nightly won't be broken. Please review when you have some availability.

zuk3975 commented on src/Core/Domain/OrderState/OrderStateSettings.php:
```diff
@@ -24,12 +24,14 @@
+    /**
+     * Order state name max length as defined in the ObjectModel
+     */
+    public const NAME_MAX_LENGTH = 64;
```
You should use this const in [OrderState](https://github.com/PrestaShop/PrestaShop/blob/develop/classes/order/OrderState.php#L94) and [OrderReturnState](https://github.com/PrestaShop/PrestaShop/blob/develop/classes/order/OrderReturnState.php#L45) name field too now so it is the only source of truth no matter if we validate form or objectModel.

Progi1984 replied:
Fixed

Progi1984 committed:
Order State : Use settings in legacy Objects

MhiriFaten reviewed (commented):
Hello @Progi1984 

I have checked your PR and I noticed the following issues : 

**First Issue :** 

1. Go to BO> Shop Parameters> Order Settings> Statuses page
2. Add new order status
3.  :warning: See exception

[Untitled_ Feb 22, 2023 3_37 PM.webm](https://user-images.githubusercontent.com/111756615/220656574-9116c6c8-8f0d-4759-9f24-ad69abd732b2.webm)

**Second Issue**

1. Go to BO> Shop Parameters> Order Settings> Statuses page
2. Edit order status
3.  :warning: See exception

[Untitled_ Feb 22, 2023 3_39 PM.webm](https://user-images.githubusercontent.com/111756615/220657018-0c7b0488-9c49-41c9-8f96-f2e7253f9507.webm)


Could you check it please ?

jolelievre commented on admin-dev/themes/default/template/controllers/statuses/helpers/form/form.tpl:
```diff
@@ -1,117 +0,0 @@
-{**
```
Sorry but it's a bit too early, usually the legacy controller and its associated templates are removed in a second PR once the migration PR has been validated and is activated.
The main reason for this (but not the only one) is to allow having both pages to compare their behaviour during the QA phase. So you should revert this deletetion (along with other legacy templates controllers and/or files)

jolelievre commented on admin-dev/themes/new-theme/js/components/grid/extension/bulk-action-checkbox-extension.ts:
```diff
@@ -38,10 +38,26 @@ export default class BulkActionCheckboxExtension {
+   *
+   * @private
+   */
+  private handleBulkActionCheckboxStatus(grid: Grid) {
```
Are you sure about this? From what I understand it disbaled the Select all checkbox until at leat one checbox is selected. But from a UX perspective it means you must first select a checkbox, then select the All checkbox, so it's one more click that may be unnecessary no?
We should be able to click on it right away, unless I'm missing something

Hlavtox replied:
@Progi1984 @jolelievre 

Yes, bulk action checkbox should be enabled all the time, like it works on all other pages and on the legacy.

However, order statuses page is a bit special, because we don't have the checkbox for undeletable statuses. On legacy page, when there is no checkboxes (only undeletable statuses), no bulk actions button is shown.

![Snímek obrazovky 2023-07-28 101341](https://github.com/PrestaShop/PrestaShop/assets/6097524/ecde439a-f0ae-4db0-982a-0e5666d28b4a)

![Snímek obrazovky 2023-07-28 101602](https://github.com/PrestaShop/PrestaShop/assets/6097524/588781f5-2a3c-4210-9f3f-f78cb635fc4c)

jolelievre commented on controllers/admin/AdminStatusesController.php:
```diff
@@ -1,771 +0,0 @@
-<?php
```
This file must not be removed yet

jolelievre commented on src/Adapter/OrderState/CommandHandler/AddOrderStateHandler.php:
```diff
@@ -72,6 +72,8 @@ private function fillOrderStateWithCommandData(OrderState $orderState, AddOrderS
         $orderState->paid = $command->isPaid();
         $orderState->delivery = $command->isDelivery();
-        $orderState->template = $command->getLocalizedTemplates();
+        if ($command->isSendEmailEnabled()) {
+            $orderState->template = $command->getLocalizedTemplates();
+        }
```
Should the value be forced to empty array in this case?
```suggestion
        $orderState->template = $command->isSendEmailEnabled() ? $command->getLocalizedTemplates() : [];
```

jolelievre commented on src/Adapter/OrderState/CommandHandler/EditOrderStateHandler.php:
```diff
@@ -107,6 +107,10 @@ private function updateOrderStateWithCommandData(OrderState $orderState, EditOrd
         if (null !== $command->isSendEmailEnabled()) {
             $orderState->send_email = $command->isSendEmailEnabled();
+
+            if ($orderState->send_email && null !== $command->getTemplate()) {
```
Same question as in the add handler, when `send_email` is forced to false, should the `$orderState->template` be forced to empty array to clean it?

jolelievre commented on src/Core/Domain/OrderReturnState/OrderReturnStateSettings.php:
```diff
@@ -24,12 +24,14 @@
  * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
  */
 
-header('Expires: Mon, 26 Jul 1997 05:00:00 GMT');
```
Careful the `index.php` should also be reverted and kept along with the removed templates

jolelievre commented on src/Core/Domain/OrderState/OrderStateSettings.php:
```diff
@@ -24,12 +24,14 @@
  * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
  */
 
-header('Expires: Mon, 26 Jul 1997 05:00:00 GMT');
```
Same warning about the helpers template here

jolelievre commented on src/PrestaShopBundle/Resources/config/routing/admin/configure/shop_parameters/order_states.yml:
```diff
@@ -5,7 +5,6 @@ admin_order_states:
     _controller: PrestaShopBundle\Controller\Admin\Configure\ShopParameters\OrderStateController::indexAction
     _legacy_controller: AdminStatuses
     _legacy_link: AdminStatuses
-    _legacy_feature_flag: order_state
```
Too early to remove this one as well, first the new page must be validated Then in a dedicated PR we will:
- remove legacy controller
- remove legacy templates
- remove `_legacy_feature_flag` attributes
- remove the feature flag from install data (and remove it in auto upgrade scripts as well)

jolelievre commented on src/PrestaShopBundle/Resources/views/Admin/Configure/ShopParameters/OrderReturnStates/Blocks/form.html.twig:
```diff
@@ -40,8 +40,8 @@
                     ~ ' '
-                    ~ 'Invalid characters: numbers and'|trans({}, 'Admin.Shopparameters.Help')
-                    ~ ' !<>,;?=+()@#"{}_$%:'
+                    ~ 'Invalid characters:'|trans({}, 'Admin.Shopparameters.Help')
```
I know it's not the scope of the PR, but this form is a bit "ol school" it uses the helpers `ps.form_group_row` and stuff All this wording should be configured in the Symfony form type and the twig template should be as simple as `form_row(orderReturnStateForm)`

If you prefer to handle this in another PR though no problem

jolelievre commented on tests/Integration/Classes/controller/AdminControllerTest.php:
```diff
@@ -152,7 +152,6 @@ public function getControllersClasses(): array
     {
         return [
             ['AdminCarriersController'],
-            ['AdminStatusesController'],
```
Too soon

jolelievre commented on tests/UI/campaigns/functional/BO/13_shopParameters/02_orderSettings/02_statuses/returnStatuses/01_CRUDOrderReturnStatus.ts:
```diff
@@ -94,7 +94,7 @@ describe('BO - Shop Parameters - Order Settings - Statuses : Create, update '
 
       const textResult = await addOrderReturnStatusPage.setOrderReturnStatus(page, createOrderReturnStatusData);
-      await expect(textResult).to.contains(statusesPage.successfulUpdateMessage);
+      await expect(textResult).to.contains(statusesPage.successfulCreationMessage);
```
About the auto tests, if you reintroduce the legacy controller it may be too soon to adapt them as well

jolelievre reviewed (commented):
Thanks @Progi1984 

Just a few comments, but the most important is that you shouldn't remove the legacy page in this PR, the migrated one must be validated and enabled in a dedicated PR after this one is merged

Progi1984 commented:
@jolelievre Sorry : But it's the latest PR for the migration like #29510;

Errors are present because the PR has been created in Feb 2022 (just one year), and rebased again, again (and again, again) and locked for a ColorPicker and rebased again. In this time a lot of things have been improved (Services calls, Form theme). So sorry if this PR is up to date (I miss some things when I rebased).

So I will fix  errors from @MhiriFaten and your feedbacks in specific PRs, and will rebase this PR after these one, check UI tests & CI are always good and will ask a new review and a new QA Check.

Progi1984 committed:
Functional Tests : Fixed functional/BO/13*/02*/02*/returnStatuses/01*

Progi1984 committed:
Functional Tests : Fixed functional/BO/13*/02*/02*/statuses/01*

Progi1984 committed:
Removed feature flag

Progi1984 commented:
@committers @prestashop-maintainers @PrestaShop/ui-tests-maintainers  Need review when you have some availability

hibatallahAouadni reviewed (commented):
Hello @Progi1984 

Thanks for your hard working, all old issues has been fixed 👏 
Unfortunately, I found a new bug in the Add/Edit Order Status 😞 
Here's the steps:

1. BO>Shop Parameters>Order Setting>Statuses
2. Click on `Add new order status` button 
3. Try to put invalid string as `Status name` in one language 
4. An error message is displayed => ✅ 
5. Try now to put at least the invalid string as `Status name` in two languages
6. See error => an exception is thrown (see the attached screen record below)

[pr-27590.webm](https://github.com/PrestaShop/PrestaShop/assets/23308427/fe380a34-ad56-4117-85cc-e7fd60c88f16)

![image](https://github.com/PrestaShop/PrestaShop/assets/23308427/bd992e54-cf96-45ef-b5c2-1e820e20d241)

Please check and feedback.

PS: please find the error log here.
[4143f7a02725bb51fb50668eb063bcaa.txt](https://github.com/PrestaShop/PrestaShop/files/12289927/4143f7a02725bb51fb50668eb063bcaa.txt)

Thanks!

Hlavtox commented:
@Progi1984 Parameters going to `trans(` are swapped somewhere.

Progi1984 commented:
@PrestaShop/committers & @hibatallahAouadni : 

* @Hlavtox has fixed the `trans` problem in  #33621 (Thanks to him).
* PR Rebased
* Auto Tests restarted : https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/6118886886

MhiriFaten reviewed (approved):
Hello @Progi1984 ,

I have checked your PR and all the old issues have been well fixed.
Automated tests are OK.
https://github.com/MhiriFaten/testing_pr/actions/runs/6122171602

It is QA approved :heavy_check_mark: 
**Thank you** :rocket: :muscle:

MatShir commented:
🥳

nicosomb commented:
Well done @Progi1984 🎉

Progi1984 commented:
Thanks. That was my last external contribution. 😘
